functests/performance/updating_profile.go

```diff
@@ -43,80 +47,91 @@ var _ = Describe("[rfe_id:28761] Updating parameters in performance profile", fu
-			"tuned.non_isolcpus": "00000009",
-			"isolcpus":           "1-2",
-			"new-argument":       "test",
-		}
-
-		By("Verifying that mcp is updated")
-		waitForMcpCondition(testutils.RoleWorkerRT, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue, timeout)
+	Context("Verify that all performance profile parameters can be updated", func() {
+		var removedKernelArgs string
 
-		By("Making changes in profile")
 		hpSize := performancev1alpha1.HugePageSize("2M")
 		isolated := performancev1alpha1.CPUSet("1-2")
 		reserved := performancev1alpha1.CPUSet("0,3")
 		policy := "best-effort"
 		f := false
 
-		profile.Spec.HugePages = &performancev1alpha1.HugePages{
-			DefaultHugePagesSize: &hpSize,
-			Pages: []performancev1alpha1.HugePage{
-				{
-					Count: 5,
-					Size:  hpSize,
-				},
-			},
-		}
-		profile.Spec.CPU = &performancev1alpha1.CPU{
-			BalanceIsolated: &f,
-			Reserved:        &reserved,
-			Isolated:        &isolated,
-		}
-		profile.Spec.NUMA = &performancev1alpha1.NUMA{
-			TopologyPolicy: &policy,
-		}
-		profile.Spec.RealTimeKernel = &performancev1alpha1.RealTimeKernel{
-			Enabled: &f,
-		}
-
-		var removedKernelArgs string
-		if profile.Spec.AdditionalKernelArgs == nil {
-			By("AdditionalKernelArgs is empty. Checking only adding new arguments")
-			profile.Spec.AdditionalKernelArgs = append(profile.Spec.AdditionalKernelArgs, "new-argument=test")
-		} else {
-			removedKernelArgs = profile.Spec.AdditionalKernelArgs[0]
-			profile.Spec.AdditionalKernelArgs = append(profile.Spec.AdditionalKernelArgs[1:], "new-argument=test")
-		}
+		It("Modify profile and verify that MCO successfully updated the node", func() {
+			// timeout should be based on the number of worker nodes
+			timeout := time.Duration(len(workerRTNodes) * mcpUpdateTimeout)
 
-		By("Applying changes in performance profile and waiting until mcp will start updating")
-		Expect(testclient.Client.Update(context.TODO(), profile)).ToNot(HaveOccurred())
-		waitForMcpCondition(testutils.RoleWorkerRT, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue, timeout)
-
-		By("Waiting when mcp finishes updates and verifying new parameters applied")
-		waitForMcpCondition(testutils.RoleWorkerRT, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue, timeout)
+			By("Modifying profile")
+			profile.Spec.HugePages = &performancev1alpha1.HugePages{
+				DefaultHugePagesSize: &hpSize,
+				Pages: []performancev1alpha1.HugePage{
+					{
+						Count: 5,
+						Size:  hpSize,
+					},
+				},
+			}
+			profile.Spec.CPU = &performancev1alpha1.CPU{
+				BalanceIsolated: &f,
+				Reserved:        &reserved,
+				Isolated:        &isolated,
+			}
+			profile.Spec.NUMA = &performancev1alpha1.NUMA{
+				TopologyPolicy: &policy,
+			}
+			profile.Spec.RealTimeKernel = &performancev1alpha1.RealTimeKernel{
+				Enabled: &f,
+			}
 
-		for _, node := range workerRTNodes {
-			checkCmdLineArgs(node, newCmdLineArgs)
+			if profile.Spec.AdditionalKernelArgs == nil {
```

Comment:
hm for me it looks like different test cases:
```golang
It("should succeed to add new kernel argument", func() {})
It("should succeed to add and remove kernel argument in the same time", func() {})
```
and the first one you can run on every environment, for the second I assume that you will update the profile under the `BeforeEach` to have at least one kernel argument, in case if it does not have any and after you will run the test.

I know that it is very time-consuming, so maybe we can introduce `BeforeAll` and make preparation once for all tests, WDT?